src/v2/components/Cell/components/Konnectable/components/KonnectableGeneric/index.tsx

```diff
@@ -25,18 +28,45 @@ interface Props extends BoxProps {
   srcs: string[]
   title?: string
   mode: Mode
+  originalDimensions?: {
+    width: number
+    height: number
+  }
 }
 
 export const KonnectableGeneric: React.FC<Props> = ({
   src,
   srcs,
   title = null,
   mode = Mode.RESTING,
+  originalDimensions,
   ...rest
-}) => (
-  <Container mode={mode} border="1px solid" borderColor="transparent" {...rest}>
-    <KonnectableImg src={src} srcs={srcs} alt={title} />
-  </Container>
-)
+}) => {
+  const { width: originalWidth, height: originalHeight } = originalDimensions
+
+  const { width, height } = scale({
+    width: originalWidth,
+    height: originalHeight,
+    maxWidth: Math.min(originalWidth, parseInt(constants.blockWidth)),
+    maxHeight: Math.min(originalHeight, parseInt(constants.blockWidth)),
```

Comment:
You can/should set the maxWidth to just be the blockWidth here. You never want it to be larger than that presumably?

Reply:
Oh I see, you don't want to scale it if it's smaller?

Reply:
Exactly